azure-pipelines.yml

```diff
@@ -482,11 +491,11 @@ stages:
 
       - template: set-azp-variables-steps.yml@pipelines-yaml-templates
 
-      - task: Maven@3
+      - task: JFrogMaven@1
         displayName: 'Maven install'
         env:
           ARTIFACTORY_DEPLOY_USERNAME: $(ARTIFACTORY_DEPLOY_USERNAME)
-          ARTIFACTORY_DEPLOY_PASSWORD: $(ARTIFACTORY_DEPLOY_PASSWORD)
+          ARTIFACTORY_DEPLOY_PASSWORD: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
```

Comment:
@elrzn TBH here I don't understand why the DEPLOY username works with the PRIVATE_READER token

Reply:
If you are using `artifactoryDeployService`, try removing `ARTIFACTORY_DEPLOY_USERNAME` and `ARTIFACTORY_DEPLOY_PASSWORD` as they seem to be mutually exclusive.

Reply:
I removed them and it fails

> [main] ERROR org.apache.maven.plugins.enforcer.EnforceMojo - Rule 4: org.apache.maven.plugins.enforcer.RequireEnvironmentVariable failed with message:
Environment variable "ARTIFACTORY_DEPLOY_PASSWORD" is required for this build.

https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=76559&view=logs&jobId=5a84c39f-87b6-5ab2-d0a7-dae390472f36&j=5a84c39f-87b6-5ab2-d0a7-dae390472f36&t=751e0046-590d-5d14-2e2f-85058296ad91